samza-core/src/main/java/org/apache/samza/operators/impl/WindowOperatorImpl.java

```diff
@@ -74,22 +78,35 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(WindowOperatorImpl.class);
 
+  private final WindowOperatorSpec<M, WK, WV> windowOpSpec;
+  private final Clock clock;
   private final WindowInternal<M, WK, WV> window;
   private final KeyValueStore<WindowKey<WK>, WindowState<WV>> store = new InternalInMemoryStore<>();
-  TriggerScheduler<WK> triggerScheduler ;
+  private TriggerScheduler<WK> triggerScheduler;
+
+  // Results to be returned for the current handleMessage and handleTimer call.
+  private List<WindowPane<WK, WV>> currentResults = new ArrayList<>();
```

Comment:
I was thinking that the version that returns `List<WindowPane>` might be slightly cleaner. But, now that I see the implementation, It's not vastly different. +0 on this one. Your pick 👍